src/main/java/spireMapOverhaul/zones/gremlincamp/monsters/GremlinCook.java

```diff
@@ -0,0 +1,146 @@
+package spireMapOverhaul.zones.gremlincamp.monsters;
+
+import com.badlogic.gdx.math.MathUtils;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.GameActionManager;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.common.DamageAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction;
+import com.megacrit.cardcrawl.actions.common.RollMoveAction;
+import com.megacrit.cardcrawl.actions.unique.IncreaseMaxHpAction;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.cards.status.Dazed;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.MonsterStrings;
+import com.megacrit.cardcrawl.powers.PoisonPower;
+import com.megacrit.cardcrawl.powers.StrengthPower;
+import com.megacrit.cardcrawl.vfx.combat.HealEffect;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.abstracts.AbstractSMOMonster;
+import spireMapOverhaul.actions.AllEnemyApplyPowerAction;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zones.gremlincamp.PlayerPoisonPower;
+
+import java.util.ArrayList;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+
+public class GremlinCook extends AbstractSMOMonster {
+    public static final String ID = makeID(GremlinCook.class.getSimpleName());
+    private static final MonsterStrings monsterStrings = CardCrawlGame.languagePack.getMonsterStrings(ID);
+    public static final String NAME = monsterStrings.NAME;
+    public static final String[] MOVES = monsterStrings.MOVES;
+
+    private static final Byte ALL_BUFF = 0, ALL_BULK = 1, POISON = 2, TACKLE = 3;
+
+    private int healAmt, strAmt, poisonAmt;
+    private boolean secondMove = false;
+
+    public GremlinCook() {
+        this(0, 0);
+    }
+
+    public GremlinCook(float x, float y) {
+        super(NAME, ID, 50, 0, 0, 120f, 210f, SpireAnniversary6Mod.makeMonsterPath("GremlinCamp/GremlinCook.png"), x, y);
+        setHp(calcAscensionTankiness(56), calcAscensionTankiness(62));
+        addMove(ALL_BUFF, Intent.BUFF);
+        addMove(ALL_BULK, Intent.BUFF);
+        addMove(POISON, Intent.STRONG_DEBUFF);
+        addMove(TACKLE, Intent.ATTACK_DEBUFF, calcAscensionDamage(10));
+
+        healAmt = calcAscensionSpecial(8);
+        strAmt = AbstractDungeon.ascensionLevel >= 18? 3: 2;
+        poisonAmt = calcAscensionSpecial(6);
+    }
+
+    @Override
+    protected void setUpMisc() {
+        super.setUpMisc();
+        this.type = EnemyType.NORMAL;
+    }
+
+    @Override
+    public void takeTurn() { //TODO: Add sounds
+        switch (nextMove) {
+            case 0:
+                useShakeAnimation(Settings.ACTION_DUR_MED);
+                Wiz.atb(new AllEnemyApplyPowerAction(this, strAmt, (m) -> new StrengthPower(m, strAmt)));
```

Comment:
The only fight where Gremlin Cook shows up has it as the last monster to act, right?

Just wanted to be sure about that since otherwise this would change enemy damage mid-turn, which the base game avoids by setting up fights so that strength buffing and vulnerable applying enemies always act last.

Reply:
Yep, it always acts last